src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -29,37 +30,56 @@ export const PartyMap: React.FC<PartyMapProps> = ({ venue }) => {
   const { user, profile } = useUser();
   const { recentVenueUsers } = useRecentVenueUsers({ venueName: venue.name });
 
-  const { relatedVenues } = useRelatedVenues({ currentVenueId: venue.id });
-
-  const selfAndChildVenueIds = useMemo(
-    () =>
-      relatedVenues
-        .filter(
-          (relatedVenue) =>
-            relatedVenue.parentId === venue.id || relatedVenue.id === venue.id
-        )
-        .map((childVenue) => childVenue.id),
-    [relatedVenues, venue]
+  const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
+
+  const hasSelectedRoom = !!selectedRoom;
+
+  const [portalVenueId] = useMemo(
+    () => getLastUrlParam(selectedRoom?.url ?? ""),
+    [selectedRoom]
   );
 
-  const { events: selfAndChildVenueEvents } = useVenueEvents({
-    venueIds: selfAndChildVenueIds,
+  const venueId = useMemo(() => (hasSelectedRoom ? portalVenueId : venue.id), [
+    venue,
+    hasSelectedRoom,
+    portalVenueId,
+  ]);
+  const { relatedVenues: selectedRoomRelatedVenues } = useRelatedVenues({
+    currentVenueId: venueId,
   });
 
-  const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
+  const selectedRoomselfAndChildVenueIds = useMemo(
+    () => selectedRoomRelatedVenues.map((childVenue) => childVenue.id),
+    [selectedRoomRelatedVenues]
+  );
 
-  const hasSelectedRoom = !!selectedRoom;
+  const { events: selectedRoomEventsSelfAndChildVenueEvents } = useVenueEvents({
```

Comment:
`events: relatedVenuesEvents`

Reply:
fixed in 3419bf47ea4120a2e14dc9ada6a10449c81c41cf